virttest/qemu_vm.py

```diff
@@ -4100,8 +4100,9 @@ def migrate(self, timeout=virt_vm.BaseVM.MIGRATE_TIMEOUT, protocol="tcp",
 
             # Switch self <-> clone
             temp = self.clone(copy_state=True)
-            self.__dict__ = clone.__dict__
-            clone = temp
+            self.destroy(gracefully=False)      # self is the source dead vm
+            self.__dict__ = clone.__dict__      # self becomes the dst vm
+            clone = temp    # for cleanup purposes keep clone
```

Comment:
The source VM and destination VM  should be exchanged only after the migration is successful. so, I think lines from 4102 to 4105  should be move to line#4133